pkg/planner/core/task.go

```diff
@@ -2235,6 +2238,9 @@ func (p *PhysicalHashAgg) attach2TaskForMpp(tasks ...base.Task) base.Task {
 			newMpp := mpp.enforceExchanger(exProp)
 			attachPlan2Task(middle, newMpp)
 			mpp = newMpp
+			if partialHashAgg, ok := partial.(*PhysicalHashAgg); ok && len(partitionCols) != 0 {
+				partialHashAgg.tiflashPreAggMode = p.SCtx().GetSessionVars().TiFlashPreAggMode
+			}
```

Comment:
For some agg, it may generate 3 stage aggregation, will the first 2 stage agg be treated as partial agg?

Reply:
I checked the 3-stage agg and found most 2nd agg has no group by column. So I think the 2nd agg no need to use auto pass through